plugins/module_utils/rest_client.py

```diff
@@ -133,7 +134,10 @@ def put_record(
             ).json
         except TimeoutError as e:
             raise errors.ScaleComputingError(f"Request timed out: {e}")
-
+        except (json.JSONDecodeError, json.decoder.JSONDecodeError) as e:
+            raise json.JSONDecodeError(e.msg, e.doc, e.pos)
+        except ValueError as e:
+            raise ValueError(e)
```

Comment:
https://github.com/ScaleComputing/HyperCoreAnsibleCollection/issues/165
I though you were also getting `JSONDecodeError`, I wasn't able to reproduce, I was only getting the `ValueError`, but I added it just in case since it was mentioned in the issue.

Reply:
Aha. I was thinking `JSONDecodeError` is being raised - it surprised me it is ValueError.